posthog/models/filters/stickiness_filter.py

```diff
@@ -0,0 +1,93 @@
+from datetime import datetime
+from typing import Any, Dict, Optional, Union
+
+from django.db.models.expressions import Value
+from django.db.models.functions.datetime import TruncDay, TruncHour, TruncMinute, TruncMonth, TruncWeek, TruncYear
+from django.http import HttpRequest
+from django.utils import timezone
+
+from posthog.constants import INTERVAL, STICKINESS_DAYS
+from posthog.models.entity import Entity
+from posthog.models.event import Event
+from posthog.models.filters.filter import Filter
+from posthog.models.team import Team
+from posthog.utils import relative_date_parse
+
+
+class StickinessFilter(Filter):
+    num_intervals: int
+    date_from: datetime
+    date_to: datetime
+    interval: str = "Day"
+    entityId: Optional[str]
+    type: Optional[str]
+    stickiness_days: int
+
+    def __init__(self, data: Optional[Dict[str, Any]] = None, request: Optional[HttpRequest] = None, **kwargs) -> None:
+        super().__init__(data, request)
+        if request:
+            data = {
+                **request.GET.dict(),
+            }
+        elif not data:
+            raise ValueError("You need to define either a data dict or a request")
+
+        team: Optional[Team] = kwargs.get("team", None)
+        if not team:
+            raise ValueError("Team must be provided to stickiness filter")
+
+        if self._date_from == "all":
+            return (
```

Comment:
Why are we returning data in a constructor? :confused:

Reply:
fantastic question..was pasting over earliest date logic

Reply:
thanks for the catch—haven't finished writing all tests yet. stickiness was severely lacking tests for most filtering